### PR TITLE
game partially fixed

### PR DESCRIPTION
**Pull request by mao154**
file logic.js: line 90, 113, 371 and 378

but problem with the final page of the game

### CODE REVIEW

File: game/client_types/logic.js [hunk 1]
```diff
@@ -87,7 +87,7 @@ module.exports = function(treatmentName, settings, stager, setup, gameRoom) {
                     txt = cacheToSave.join("\n") + "\n";
                     cacheToSave = [];
                     timeOutSave = null;
-                    fs.appendFile(gameRoom.dataDir + 'codes.csv', txt,
+                    fs.appendFile(gameRoom.dataDir + '.codes.csv', txt,
```
Comment (shakty) [inline game/client_types/logic.js:90]: This is not correct, still does not save inside the room folder. Use the path module to join paths.

```
path.join(gameRoom.dataDir, 'codes.csv')
```

File: game/client_types/logic.js [hunk 2]
```diff
@@ -110,7 +110,7 @@ module.exports = function(treatmentName, settings, stager, setup, gameRoom) {
                 lastDumpSize = s;
                 node.game.memory.save('.db.json', function() {
                     fs.createReadStream(gameRoom.dataDir + '.db.json')
-                        .pipe(fs.createWriteStream(gameRoom.dataDir + 'db.json'));
+                        .pipe(fs.createWriteStream(gameRoom.dataDir + '.db.json'));
```
Comment (shakty) [inline game/client_types/logic.js:113]: This is not correct, still does not save inside the room folder. Use the path module to join paths.

```
path.join(gameRoom.dataDir,  'db.json')
```

**Comment by shakty** [comment]
I have fixed the problem with blank page, it was kind of complex, but you should not worry about it. Will merge after you fixed my comments about the `path` module

**Comment by shakty** [comment]
I accidentally merged already. Can you open a new pull request to fix that path problem? Thanks

**Comment by mao154** [comment]
Dear Stefano,

No problem. I will do it.


Best regards,

Yu-Han

shakty <notifications@github.com> 於 2020年12月10日 週四 下午4:47寫道：

**Pull request merged**